### PR TITLE
Initial Angular code bring up for nimble-rich-text-viewer component

### PR DESCRIPTION
**Pull request by vivinkrishna-ni**
# Pull Request

## 🤨 Rationale

This PR includes the angular integration for the `nimble-rich-text-viewer` component, and it is intended to make the component available to the Angular application which allows us to get started with the other features of comments dependent on this component.

## 👩‍💻 Implementation

* Added directive and module for `nimble-rich-text-viewer`. 
* Added `nimble-rich-text-viewer` to the example client app.

**Note**: We will add the input properties and attributes once the support for the markdown parser and other functionality in the component ( https://github.com/ni/nimble/pull/1335) is completed.

## 🧪 Testing

Added basic unit tests for the directive. 

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: angular-workspace/projects/ni/nimble-angular/src/directives/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts
Comment (jattasNI) [inline angular-workspace/projects/ni/nimble-angular/src/directives/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts:6]: If you take a look at [the spec for some other directives](https://github.com/ni/nimble/blob/main/angular-workspace/projects/ni/nimble-angular/src/directives/button/tests/nimble-button.directive.spec.ts) you can see the pattern we've established for testing them. It includes testing each property in 3 different `describe` blocks for 3 different scenarios. Could you try to copy that pattern for this component?

File: angular-workspace/projects/ni/nimble-angular/src/directives/rich-text-viewer/nimble-rich-text-viewer.directive.ts
Comment (jattasNI) [inline angular-workspace/projects/ni/nimble-angular/src/directives/rich-text-viewer/nimble-rich-text-viewer.directive.ts:1]: We are in the very early stages of reorganizing nimble-angular to expose a separate secondary entry point for each component. This work is tracked by #172 and was started for just the table component in #1252. I think we should attempt to follow the new layout for this new component rather than having to move it later (a breaking change) when we complete work on that issue.

Could you try following the layout from that PR's description? If we run into major issues we could debate doing this now but I think it'll save time in the long run to start doing it from the beginning.

File: angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.html
Comment (vivinkrishna-ni) [reply angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.html]: Moved

File: angular-workspace/projects/ni/nimble-angular/src/directives/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts
Comment (vivinkrishna-ni) [reply angular-workspace/projects/ni/nimble-angular/src/directives/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts]: Yes, I have written the tests in [this branch](https://github.com/ni/nimble/blob/users/vivin/update-angular-directive-for-rich-text-viewer/angular-workspace/projects/ni/nimble-angular/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts) for the API and will create a separate PR once the https://github.com/ni/nimble/pull/1335 is completed

We have decided to raise this PR now to include `nimble-rich-text-viewer` in the wrapper component planned for the comments feature in SLE and create separate PRs whenever we update the component. This will allow us to begin working with the actual `nimble-rich-text-viewer` importing from `@ni/nimble-angular`.

File: angular-workspace/projects/ni/nimble-angular/src/directives/rich-text-viewer/nimble-rich-text-viewer.directive.ts
Comment (vivinkrishna-ni) [reply angular-workspace/projects/ni/nimble-angular/src/directives/rich-text-viewer/nimble-rich-text-viewer.directive.ts]: Thanks for letting us know. We are unaware of this new layout to expose a secondary entry point for each component. I changed it into a new layout and verified the working of the directive in the example-client-app. Please review the changes and let me know your comments.

Comment (jattasNI) [reply angular-workspace/projects/ni/nimble-angular/src/directives/rich-text-viewer/nimble-rich-text-viewer.directive.ts]: The new entry point looks good to me. Thanks for making that change.

File: angular-workspace/projects/ni/nimble-angular/src/directives/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts
Comment (jattasNI) [reply angular-workspace/projects/ni/nimble-angular/src/directives/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts]: Excellent, I'm glad you've got those tests written already. Our general preference is to submit the tests with their corresponding code since it makes it easier to track that no test coverage gets missed. So for this PR that would mean either:
1. copy the full contents of `nimble-rich-text-viewer.directive.spec.ts` from your other branch to this one
2. remove the `markdownValue` and `fitToContent` properties from the directive in this PR and add them in the follow-on PR

I may be missing something that makes those options challenging but if it's as simple as moving some files between branches then let's default to keeping the code and tests in sync at all times.

Comment (vivinkrishna-ni) [reply angular-workspace/projects/ni/nimble-angular/src/directives/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts]: Sure, I have removed the properties and had only the relevant test cases covered for bringing up the angular code. I will update it in the follow on PRs.

File: angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.html
Comment (vikisekarNI) [inline angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.html:175]: I understand that we are yet to implement `markdownValue` property inputs inside `nimble-components` and `nimble-angular`. With this, do we still need to send an non-existing input `markdownValue` already in this PR, which will be anyway ignored for rendering? 

Let me know if we will update this PR after #1335.

After the `markdownValue` property setup, consider sending a more representative markdown input similar to `I am a rich text viewer. I can help you format text in **Bold**, _Italics_,...` as an input.

File: angular-workspace/projects/ni/nimble-angular/rich-text-viewer/nimble-rich-text-viewer.directive.ts
Comment (vikisekarNI) [inline angular-workspace/projects/ni/nimble-angular/rich-text-viewer/nimble-rich-text-viewer.directive.ts:14]: This class needs to be updated with `@Input` for `markdownValue`?

File: angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.html
Comment (vivinkrishna-ni) [reply angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.html]: I have removed the `markdownValue` property from the example-client-app. 

Yes, we will update it once https://github.com/ni/nimble/pull/1335 is complete.

File: angular-workspace/projects/ni/nimble-angular/rich-text-viewer/nimble-rich-text-viewer.directive.ts
Comment (vivinkrishna-ni) [reply angular-workspace/projects/ni/nimble-angular/rich-text-viewer/nimble-rich-text-viewer.directive.ts:14]: Yes, it will be updated in the separate PR once https://github.com/ni/nimble/pull/1335 is complete. 

This implementation is done in [this branch](https://github.com/ni/nimble/blob/users/vivin/update-angular-directive-for-rich-text-viewer/angular-workspace/projects/ni/nimble-angular/rich-text-viewer/nimble-rich-text-viewer.directive.ts#L15) already.

**Review by vikisekarNI — APPROVED** [review]

File: angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.html
Comment (jattasNI) [inline angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.html:175]: Once you add the markdown Input, could you give this some markdown content so that it's not completely empty?

**Review by jattasNI — APPROVED** [review]

**Review by rajsite — APPROVED** [review]

File: angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.html
Comment (vivinkrishna-ni) [reply angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.html:175]: Yes, I will be adding the `markdown` string to this example client app once the viewer component is completed.

**Comment by rajsite** [comment]
@atmgrifter00 and @suseendran-ni this PR is waiting on y'alls reviews

**Review by atmgrifter00 — APPROVED** [review]

**Comment by vivinkrishna-ni** [comment]
> @atmgrifter00 and @suseendran-ni this PR is waiting on y'alls reviews

@rajsite, FYI @suseendran-ni is out of office and he'll be back only by next week. Can we bypass his review as other code owners approved it?

**Comment by rajsite** [comment]
> @atmgrifter00 and @suseendran-ni this PR is waiting on y'alls reviews

yep, their review was not blocking, notice that the conditions were met. If you are confident that it is possible to merge you are able to do so, no override should be needed. I merged from main and turned on autocomplete but in the future if it's reasy to merge and you have required approvals that you are aware of you can merge.

**Pull request merged**
